Some sanity checks are failing:
```
lib/ansible/modules/cloud/docker/docker_swarm_service.py:0:0: E305 DOCUMENTATION.options.mounts.tmpfs_mode: extra keys not allowed @ data['options']['mounts']['tmpfs_mode']. Got {'description': ['File mode of the tmpfs in octal.', 'Can only be used when I(mode) is C(tmpfs).'], 'type': 'int', 'version_added': '2.8'}

lib/ansible/modules/cloud/docker/docker_swarm_service.py:0:0: E305 DOCUMENTATION.options.mounts.tmpfs_size: extra keys not allowed @ data['options']['mounts']['tmpfs_size']. Got {'description': ['Size of the tmpfs mount (format: C(<number>[<unit>])). Number is a positive integer. Unit can be C(B) (byte), C(K) (kibibyte, 1024B), C(M) (mebibyte), C(G) (gibibyte), C(T) (tebibyte), or C(P) (pebibyte).', 'Can only be used when I(mode) is C(tmpfs).'], 'type': 'str', 'version_added': '2.8'}
```

--- a/lib/ansible/modules/cloud/docker/docker_swarm_service.py
+++ b/lib/ansible/modules/cloud/docker/docker_swarm_service.py
@@ -2274,9 +2274,9 @@ def _detect_mount_tmpfs_usage(client):
     for mount in client.module.params['mounts'] or []:
         if mount.get('type') == 'tmpfs':
             return True
-        if mount.get('tmpfs_size'):
+        if mount.get('tmpfs_size') is not None:
             return True
-        if mount.get('tmpfs_mode'):
+        if mount.get('tmpfs_mode') is not None:
             return True
     return False
 

--- a/lib/ansible/modules/cloud/docker/docker_swarm_service.py
+++ b/lib/ansible/modules/cloud/docker/docker_swarm_service.py
@@ -338,20 +338,20 @@ options:
             type: dict
         type: dict
         version_added: "2.8"
-    tmpfs_size:
-      description:
-        - "Size of the tmpfs mount (format: C(<number>[<unit>])). Number is a positive integer.
-          Unit can be C(B) (byte), C(K) (kibibyte, 1024B), C(M) (mebibyte), C(G) (gibibyte),
-          C(T) (tebibyte), or C(P) (pebibyte)."
-        - Can only be used when I(mode) is C(tmpfs).
-      type: str
-      version_added: "2.8"
-    tmpfs_mode:
-      description:
-        - File mode of the tmpfs in octal.
-        - Can only be used when I(mode) is C(tmpfs).
-      type: int
-      version_added: "2.8"
+      tmpfs_size:
+        description:
+          - "Size of the tmpfs mount (format: C(<number>[<unit>])). Number is a positive integer.
+            Unit can be C(B) (byte), C(K) (kibibyte, 1024B), C(M) (mebibyte), C(G) (gibibyte),
+            C(T) (tebibyte), or C(P) (pebibyte)."
+          - Can only be used when I(mode) is C(tmpfs).
+        type: str
+        version_added: "2.8"
+      tmpfs_mode:
+        description:
+          - File mode of the tmpfs in octal.
+          - Can only be used when I(mode) is C(tmpfs).
+        type: int
+        version_added: "2.8"
   name:
     description:
       - Service name.

--- a/test/integration/targets/docker_swarm_service/tasks/tests/mounts.yml
+++ b/test/integration/targets/docker_swarm_service/tasks/tests/mounts.yml
@@ -1,6 +1,6 @@
 - name: Registering service name
   set_fact:
-    service_name: "{{ name_prefix ~ '-logging' }}"
+    service_name: "{{ name_prefix ~ '-mounts' }}"
     volume_name_1: "{{ name_prefix ~ '-volume-1' }}"
     volume_name_2: "{{ name_prefix ~ '-volume-2' }}"
 

